model/roles.go

```diff
@@ -134,22 +145,154 @@ func LoadRoleManifest(manifestFilePath string, releases []*Release) (*RoleManife
 	return &rolesManifest, nil
 }
 
+// SetGlobalConfig initializes the roleManifest's configuration of global entities
+// that it needs later to calculate the dev version:
+// the global properties, processed light opinions (to get the properties), and raw dark opinions
+func (m *RoleManifest) SetGlobalConfig(lightManifestPath, darkManifestPath string) error {
+	manifestConfig := manifestConfiguration{}
+	opinions, err := manifestConfig.loadOpinionsFromPath(lightManifestPath)
+	if err != nil {
+		return err
+	}
+	manifestConfig.lightOpinions = opinions
+	opinions, err = manifestConfig.loadOpinionsFromPath(darkManifestPath)
+	if err != nil {
+		return err
+	}
+	manifestConfig.darkOpinions = opinions
+	manifestConfig.globalDefaultValues = getDefaultValuesHash(m.Configuration.Templates)
+	m.globalConfig = &manifestConfig
+	return nil
+}
+
 // GetRoleManifestDevPackageVersion gets the aggregate signature of all the packages
 func (m *RoleManifest) GetRoleManifestDevPackageVersion(extra string) string {
 	// Make sure our roles are sorted, to have consistent output
 	roles := append(Roles{}, m.Roles...)
 	sort.Sort(roles)
 
 	hasher := sha1.New()
-	hasher.Write([]byte(extra))
+	io.WriteString(hasher, extra)
 
 	for _, role := range roles {
-		hasher.Write([]byte(role.GetRoleDevVersion()))
+		s, err := role.GetRoleDevVersion()
+		if err == nil {
```

Comment:
This code is obsolete.  GetRoleDevVersion will be rewritten to just return a string.